org.code.toboggan.filesystem/src/org/code/toboggan/filesystem/extensions/file/FSFileCreate.java

```diff
@@ -0,0 +1,54 @@
+package org.code.toboggan.filesystem.extensions.file;
+
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.Set;
+
+import org.code.toboggan.core.CoreActivator;
+import org.code.toboggan.core.extension.APIExtensionIDs;
+import org.code.toboggan.core.extension.AbstractExtensionManager;
+import org.code.toboggan.core.extension.ICoreExtension;
+import org.code.toboggan.filesystem.extensionpoints.file.IFSFileCreateExt;
+import org.code.toboggan.filesystem.extensions.FileSystemExtensionManager;
+import org.code.toboggan.network.request.extensionpoints.file.IFileCreateResponse;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.ResourcesPlugin;
+
+import clientcore.dataMgmt.FileController;
+import clientcore.dataMgmt.SessionStorage;
+import clientcore.websocket.models.File;
+
+public class FSFileCreate implements IFileCreateResponse {
+
+	private SessionStorage ss;
+	private FileController fc;
+	private AbstractExtensionManager extMgr;
+	
+	public FSFileCreate() {
+		this.ss = CoreActivator.getSessionStorage();
+		this.fc = new FileController(ss);
+		this.extMgr = FileSystemExtensionManager.getInstance();
+	}
+	
+	@Override
+	public void fileCreated(long fileID, String name, Path absolutePath, String projectRelativePath, long projectID) {		
+		// NOTE: This is only here for now so that I don't forget to include it when making extensions in the model plugin
+		File f = new File(fileID, name, projectRelativePath, 1, ss.getUsername(), new Date().toString()); // doing date like this is questionable, but it's not being used
+		ss.setFile(projectID, f);
```

Comment:
There's a comment above it that explains that why it's here and not in the models plugin, but to further elaborate, the way this is done in the original plugin is that the project creation call waits with a Semaphore for all of the files to finish and then does a GetFiles request to create the file metadata and set it for the project metadata all at once. The reason I put this in here was because I wanted to make sure that I would remember to actually create the file models when transferring model creation to the plugin because in the old plugin, this was intertwined with the project creation code in a weird way.